latex-syntax/src/main/java/org/xwiki/contrib/latex/internal/FigureTool.java

```diff
@@ -42,4 +49,60 @@
      * @return true if the figure caption macro is the last content inside the figure macro content or false otherwise
      */
     boolean isFigureCaptionLast(Block figureCaptionBlock);
+
+    /**
+     * Returns the environment to use for the current caption. By default, "table" or "figure" are returned but
+     * extensions can allow for different environments. Note that different environments mights require different
+     * parameters (see {@link #getFigureParameter(FigureBlock)}), or might not support captions in their content (see
+     * {@link #displayFigureCaption(FigureCaptionBlock)}).
+     *
+     * @param figureBlock the figure block for which to get the environment name
+     * @return the name of the environment (e.g., "figure", "block", or another value in case of customization by an
+     *     extension)
+     * @since 1.21
+     */
+    @Unstable
+    default String getFigureEnvironment(Block figureBlock)
+    {
+        return isTable(figureBlock) ? "table" : "figure";
+    }
+
+    /**
+     * Return the figure parameter for a figure block. The figure parameter must be synchronized with the returned
+     * figure environment (see {@link #getFigureEnvironment(Block)}. For instance, for the figure environments "figure"
+     * or "table", the default parameters is {@code "h"} (setting the figure location to "here"), resulting in the
+     * following latex.
+     * <pre>
+     * \begin{figure}[h]
+     * % Figure content.
+     * \end{figure}
+     * </pre>
+     *
+     * @param figureBlock the figure block for which to generate the parameters
+     * @return the parameter of the figure (i.e., {@code "h"} by default, but can be overridden by extension)
+     * @since 1.21
+     */
+    @Unstable
+    default Block getFigureParameter(FigureBlock figureBlock)
```

Comment:
It's not very clear what parameter is being referred to here in the method name. I don't understand why you  need this method and the fact that it's hard to describe is maybe a sign that the abstraction is not at the right level. Why not have a method that returns the LaTeX command to use to start a figure environment?

Reply:
Indeed having two methods `startEnvironment` and `endEnvironment` returning the `\begin{...}[...]` and `\end{...}` parts could be a valid option.
It's allowing to override with more flexibility, but requires slightly more work as the full start/end must be returned even though in some case only the environment or the parameter needs to be overridden.

Reply:
> Basically I don't understand the current need since for inline figure types, the template below will not work as it uses `\begin{<environmentName>}`.

Not sure to get what wouldn't work.
As long as a `\newtheorem{envName}{envLabel}` is declared in the preamble, using `\begin{envName}...\end{envName}` will work correctly.

Reply:
ah yes, I forgot that it was done in 2 steps and that the second step was using`\begin{envName}` similar to figures.

Reply:
> Indeed having two methods startEnvironment and endEnvironment returning the \begin{...}[...] and \end{...} parts could be a valid option.

Note that we may not need the closing part. We could also pass the current block as parameter to that it does the full rendering.

In any case, I'm not against keeping the `getFigureParameter` method but it needs a better name IMO to indicate what parameter we're talking about. I know understand it's the environment parameter, so maybe`getFigureEnvironmentParameter`.